src/com/carolinarollergirls/scoreboard/core/impl/TeamImpl.java

```diff
@@ -298,10 +288,13 @@ public TeamSnapshot snapshot() {
     public void restoreSnapshot(TeamSnapshot s) {
         synchronized (coreLock) {
             if (s.getId() != getId()) {	return; }
-            for (ValueWithId skater : getAll(Child.SKATER)) {
-                ((Skater)skater).restoreSnapshot(s.getSkaterSnapshot(skater.getId()));
-            }
+            boolean ended = s.getFieldingAdvancePending();
+            set(Value.FIELDING_ADVANCE_PENDING, ended);
             updateTeamJams();
+            if (scoreBoard.isInJam()) {
+                set(Value.CALLOFF, false);
```

Comment:
This doesn't feel right, shouldn't we restore rather than set?

E.g. start jam -> lead -> unstart jam -> start jam incorrectly has lead set on the new jam.

Reply:
Call and Inj: If the Jam is ended via the Calloff or Injury button, the field will be set before the snapshot is taken. And there are no circumstances where setting either before the Jam has ended is correct.

Lead and Lost: If a Jam has gone on long enough that one of those has been marked, there's a significant chance that actually the undo is in error. And if the operator cancels the undo (i.e replace Start Jam with Start Jam), it'd be surprising if the marks were dropped, especially as scoring trips or SP aren't dropped either. (And I'd be vehemently opposed to removing trips and thus possibly points on a clock action.)

I have addressed the other remarks.